analyzers/tests/SonarAnalyzer.UnitTest/MetadataReferences/NuGetMetadataFactory.Package.cs

```diff
@@ -45,149 +50,71 @@ public Package(string id, string version, string runtime)
-            /// Package directory names are in the form "{package id}.{package version}".
-            /// The list is sorted in ascending order, so the most recent version will be last.
-            /// </remarks>
-            private IEnumerable<string> SortedPackageFolders()
-            {
-                // The package will be in a folder called "\packages\{packageId}.{version}", but:
-                // : the package might not be installed
-                // : there might be multiple versions installed
-                // : there might be a package that starts with the same package id
-                //      e.g. Microsoft.AspNetCore.Core and Microsoft.AspNetCore.Core.Diagnostics
-                // Most packages have a three-part version, but some have four. We don't check
-                // the actual number of parts, as long as there is at least one.
-                var matcher = new Regex($@"{Regex.Escape(Id)}(\.\d+)+$", RegexOptions.IgnoreCase);
-
-                return Directory.Exists(PackagesFolder)
-                    ? Directory.GetDirectories(PackagesFolder, $"{Id}.*", SearchOption.TopDirectoryOnly).Where(x => matcher.IsMatch(x)).OrderBy(x => x)
-                    : Enumerable.Empty<string>();
-            }
+                using var packageStream = new MemoryStream();
+                await resource.CopyNupkgToStreamAsync(
+                    Id,
+                    version,
+                    packageStream,
+                    context,
+                    NullLogger.Instance,
+                    CancellationToken.None);
 
-            private bool IsCheckForLatestPackageRequired()
-            {
-                // Install new nugets only once per day to improve the performance when running tests locally.
-                var nextCheck = NextCheckFilePath() is { } filePath && File.Exists(filePath) && DateTime.TryParse(File.ReadAllText(filePath), out var timestamp)
-                        ? timestamp
-                        : DateTime.MinValue;
-                LogMessage($"Next check for latest NuGets: {nextCheck}");
-                return nextCheck < DateTime.Now;
-            }
+                var packageDirectory = PackageDirectory();
+                using var packageReader = new PackageArchiveReader(packageStream);
 
-            private void WriteNextCheckTime()
-            {
-                const int VersionCheckDays = 5;
-                var filePath = NextCheckFilePath();
-                if (filePath == null)
+                foreach (var dllFile in packageReader.GetFiles().Where(f => f.EndsWith(".dll", StringComparison.OrdinalIgnoreCase)))
                 {
-                    return;
+                    packageReader.ExtractFile(dllFile, $"{packageDirectory}\\{dllFile}", NullLogger.Instance);
                 }
-                File.WriteAllText(filePath, DateTime.Now.AddDays(VersionCheckDays).ToString("yyyy-MM-dd"));
             }
 
-            private string NextCheckFilePath()
```

Comment:
sonar-security has a lightweight version of this because it didn't need to use the latest version. 

We need to preserve this logic, because "LATEST" on one day might not be the same version as "LATEST" one week later on the local dev build machine.